src/main/java/com/internship/backend/service/ReservationService.java

```diff
@@ -39,7 +38,23 @@ public List<Reservation> getAllReservations(){
         return reservationRepository.findAll();
     }
 
-    public Reservation addReservation(Reservation reservation) throws ReservationAlreadyExists {
+    public Reservation addReservation(Reservation reservation) throws ReservationAlreadyExists, InvalidDateException {
+
+        if (reservation.getStartTime().getMonth() != reservation.getEndTime().getMonth()) {
+            throw new InvalidDateException("The start date and end date must be within the same month");
+        }
+
+        if(reservation.getStartTime().getDay() != reservation.getEndTime().getDay()){
+            throw new InvalidDateException("The start date and end date must be within the same day");
+        }
+
+        if(reservation.getStartTime().getYear() != reservation.getEndTime().getYear()){
+            throw new InvalidDateException("The start date and end date must be within the same year");
+        }
+
+        if(reservation.getEndTime().getHour() - reservation.getStartTime().getHour() != 2){
+            throw new InvalidDateException("The reservation duration must be exactly 2 hours");
+        }
```

Comment:
This checks can be defined in a dedicated method to avoid adding complexity for this one.